### PR TITLE
Add active subscriptions intercom property

### PR DESCRIPTION
**Pull request by tspenov**
## Changes

Add active subscriptions Intercom property

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/sanbase/auth/statistics.ex [hunk 1]
```diff
@@ -116,6 +118,58 @@ defmodule Sanbase.Auth.Statistics do
     |> Repo.one()
   end
 
+  # Resource coud be watchlist, insight, user_trigger struct or any other struct which belongs to User
```
Comment (IvanIvanoff) [inline lib/sanbase/auth/statistics.ex:121]: typo coud -> could

File: lib/sanbase/auth/statistics.ex [hunk 2]
```diff
@@ -116,6 +118,58 @@ defmodule Sanbase.Auth.Statistics do
...
+      Enum.reduce(queries, resource_query, fn query_func, acc ->
+        query_func.(acc)
+      end)
+
+    query
+    |> Repo.all()
+    |> Enum.into(%{})
+  end
+
+  def screeners_user_count_map do
```
Comment (IvanIvanoff) [inline lib/sanbase/auth/statistics.ex:143]: wouldn't `user_screeners_count_map` be better?

File: lib/sanbase/auth/statistics.ex [hunk 3]
```diff
@@ -116,6 +118,58 @@ defmodule Sanbase.Auth.Statistics do
...
+      queries: [
+        fn query ->
+          from(r in query, where: fragment("?.function->>'name' != 'empty'", r))
+        end
+      ]
+    )
+  end
+
+  def active_subscriptions_map() do
+    from(s in Subscription, where: s.status == "active", preload: [plan: [:product]])
```
Comment (IvanIvanoff) [inline lib/sanbase/auth/statistics.ex:154]: Let's make this a separate function in the Subscription module - query/Repo calls from outside the schema module should be avoided.

File: lib/sanbase/auth/statistics.ex [hunk 4]
```diff
@@ -116,6 +118,58 @@ defmodule Sanbase.Auth.Statistics do
...
+    |> Sanbase.Repo.all()
+    |> Enum.map(fn s -> %{user_id: s.user_id, sub: "#{s.plan.product.name}/#{s.plan.name}"} end)
```
Comment (IvanIvanoff) [inline lib/sanbase/auth/statistics.ex:156]: `product_name` instead of `sub` would be more understandable?

File: lib/sanbase/auth/statistics.ex [hunk 5]
```diff
@@ -116,6 +118,58 @@ defmodule Sanbase.Auth.Statistics do
...
+    |> Enum.group_by(fn s -> s.user_id end)
+    |> Enum.map(fn {user_id, subs} ->
+      {user_id, Enum.map(subs, & &1.sub) |> Enum.join(",")}
```
Comment (IvanIvanoff) [inline lib/sanbase/auth/statistics.ex:159]:
```elixir
Enum.join(", ")
```
Let's add one interval for better formatting.

Comment (IvanIvanoff) [reply lib/sanbase/auth/statistics.ex:159]: One idea: You can do all the 3 steps in 1 pass (not sure if it's clearer):
```elixir
|> Enum.reduce(%{}, fn {user_id, sub}, acc ->                             
  Map.update(acc, user_id, sub, & "#{&1}, #{sub}")  
end)
```

Comment (tspenov) [reply lib/sanbase/auth/statistics.ex]: I'll go with the current version here - the transformations seem more clear to me

**Pull request merged**